src/coreclr/gc/gc.cpp

```diff
@@ -14615,8 +14671,7 @@ BOOL gc_heap::soh_try_fit (int gen_number,
                          ((res == heap_segment_mem (seg)) ||
                           ((res >= (region_mid - size)) && (res < (region_mid + size)))))
                     {
-                        HndAssignHandle(pinning_handles_for_alloc[ph_index_per_heap], 
-                            ObjectToOBJECTREF ((Object*)(acontext->alloc_ptr)));
+                        HndAssignHandleGC(pinning_handles_for_alloc[ph_index_per_heap], acontext->alloc_ptr);
```

Comment:
Nit: you could use the "res" variable here that you assigned a couple of lines above.

Reply:
indeed, I'll make the change.